src/components/lib/crt/crt.c

```diff
@@ -758,13 +805,23 @@ crt_sinv_create(struct crt_sinv *sinv, char *name, struct crt_comp *server, stru
 		.s_fn_addr   = s_fn_addr
 	};
 
-	if (crt_ns_vas_shared(client, server))
-		sinv->sinv_cap = cos_sinv_alloc(cli, srv->comp_cap_shared, sinv->s_fn_addr, client->id);
-	else
-		sinv->sinv_cap = cos_sinv_alloc(cli, srv->comp_cap, sinv->s_fn_addr, client->id);
-
+	comp_s = (srv->comp_cap_shared) ? srv->comp_cap_shared : srv->comp_cap;
+	sinv->sinv_cap = cos_sinv_alloc(cli, comp_s, sinv->s_fn_addr, client->id);
 	assert(sinv->sinv_cap);
-	printc("sinv %s cap %ld\n", name, sinv->sinv_cap);
+
+	if (crt_ns_vas_shared(client, server) && s_altfn_addr && c_fast_callgate_addr) {
+		/* values set for debugging; we need to implement a CSPRNG */
+		u64_t client_auth_tok = 0xfefefefefefefefe; /* = CSPRNG() */
+		u64_t server_auth_tok = 0xabababababababab; /* = CSPRNG() */
+
+		vaddr_t callgate_off  = c_fast_callgate_addr - sinv->client->ro_addr;
+		vaddr_t callgate_addr = (vaddr_t)sinv->client->mem + callgate_off;
+
+		jitutils_jitcallgate(callgate_addr, s_altfn_addr, sinv->client->protdom, sinv->server->protdom, client_auth_tok, server_auth_tok, client->id, sinv->sinv_cap);
```

Comment:
I agree with @gparmer that this name seems to be a little broad. It's difficult to know what the jit actually does. The name here shoule to be able to show some functionality of the function.